### PR TITLE
Fix clang warnings for transformation tests

### PR DESCRIPTION
**Pull request by olpipi**
### Details:
 - Fix clang warnings for ov_transfomations_tests

### Tickets:
 - CVS-99423

### CODE REVIEW

File: src/common/transformations/tests/common_optimizations/eliminate_duplicate_ti_inputs.cpp [hunk 1]
```diff
@@ -69,7 +69,7 @@ TEST(TransformationTests, EliminateDuplicateTIInputs) {
 
     shared_ptr<TensorIterator> ti_after_transformation;
     for (const auto& op : model->get_ordered_ops()) {
-        if (ti_after_transformation = dynamic_pointer_cast<TensorIterator>(op)) {
+        if ((ti_after_transformation = dynamic_pointer_cast<TensorIterator>(op))) {
```
Comment (ilyachur) [inline src/common/transformations/tests/common_optimizations/eliminate_duplicate_ti_inputs.cpp:72]: Why do we need this additional brackets?

**Review by ilyachur — APPROVED** [review]

File: src/common/transformations/tests/common_optimizations/eliminate_duplicate_ti_inputs.cpp (hunk 1 above)
Comment (olpipi) [reply src/common/transformations/tests/common_optimizations/eliminate_duplicate_ti_inputs.cpp:72]: clang wants it

**Pull request merged**